src/main/java/Duke.java

```diff
@@ -63,53 +70,176 @@ public static void main(String[] args) {
      *
      * @param input input given by the user.
      */
-    public static void parseInput(String input) {
+    private static void parseInput(String input) {
         System.out.println(LINES);
-        boolean error = false;
 
         if (input.equalsIgnoreCase("Bye")) {
             System.out.println(GOODBYE);
-            done = true;
+            setDone();
         } else if (input.equalsIgnoreCase("List")) {
             printList();
-        } else if (Pattern.matches("^done \\d+$", input.toLowerCase())) {
-            String[] parts = input.split(" ");
-            int index = Integer.parseInt(parts[1]) - 1;
-            if (index < listIndexTracker) {
-                list[index].markAsDone();
-                System.out.println(TAB + "Nice! I've marked this task as done:");
-                System.out.format(TASKFORMAT + "[%s] %s%n", list[index].getStatusIcon(), list[index].getDescription());
-            } else {
-                error = true;
-            }
+        } else if (isDoneInput(input)) {
+            int index = getDoneIndex(input);
+            list[index].markAsDone();
+            System.out.print(TAB + "Nice! I've marked this task as done:" + System.lineSeparator() + TAB + TAB);
+            System.out.println(list[index]);
+        } else if (isValidTodo(input)) {
+            addToList(input, TODO);
+            printAddedMessage();
+        } else if (isValidDeadline(input)) {
+            addToList(input, DEADLINE);
+            printAddedMessage();
+        } else if (isValidEvent(input)) {
+            addToList(input, EVENT);
+            printAddedMessage();
         } else {
-            addToList(input);
-            System.out.print(TAB + "added: ");
-            System.out.println(input);
+            setError(true);
         }
         if (error) {
             System.out.println(TAB + "Invalid input");
+            setError(false);
         }
         System.out.println(LINES);
     }
 
+
     /**
-     * This function adds the input string into list
+     * This function adds the input description into list accordingly, based on its task type.
      *
      * @param input input given by the user.
+     * @param taskType task type identified beforehand (DEADLINE/EVENT/TODO).
      */
-    public static void addToList(String input) {
-        list[listIndexTracker] = new Task(input);
+    private static void addToList(String input, String taskType) {
+        String[] parameters = new String[2];
```

Comment:
The variable parameters could have been named better. Not sure what it refers to.